src/warm-storage/service.ts

```diff
@@ -141,6 +143,31 @@ export class WarmStorageService {
     return this._warmStorageContract
   }
 
+  /**
+   * Get cached Warm Storage View contract instance or create new one
+   * Fetches the view contract address from the main contract if not cached
+   */
+  private async _getWarmStorageViewContract (): Promise<ethers.Contract> {
+    // Fetch view contract address if not cached
+    if (this._warmStorageViewAddress == null) {
+      const warmStorageContract = this._getWarmStorageContract()
+      this._warmStorageViewAddress = await warmStorageContract.viewContractAddress() as string
+      if (this._warmStorageViewAddress == null || this._warmStorageViewAddress === ethers.ZeroAddress || this._warmStorageViewAddress === '') {
+        throw new Error('View contract address not set in Warm Storage contract')
+      }
+    }
+
+    // Create view contract instance if not cached
+    if (this._warmStorageViewContract == null) {
+      this._warmStorageViewContract = new ethers.Contract(
+        this._warmStorageViewAddress,
+        CONTRACT_ABIS.WARM_STORAGE_VIEW,
+        this._provider
+      )
+    }
+    return this._warmStorageViewContract
+  }
```

Comment:
why not hard code the view contract address?

Reply:
We may do that in mainnet, but for now it's changeable via upgrades and supplying two contract addresses while developing this against new deploys is really cumbersome. So for now, one entry point resolves to the things we need. It also has me wondering if we shouldn't resolve the other two from WarmStorage as well ..